jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -663,6 +722,35 @@ ecma_op_object_put (ecma_object_t *object_p, /**< the object */
       }
       break;
     }
+    case ECMA_OBJECT_TYPE_CLASS:
+    {
+#ifndef CONFIG_DISABLE_TYPEDARRAY_BUILTIN
+      /* ES2015 9.4.5.5 */
+      if (ecma_is_typedarray (ecma_make_object_value (object_p)))
+      {
+        ecma_number_t num = ecma_string_to_number (property_name_p);
+        ecma_string_t *num_to_str = ecma_new_ecma_string_from_number (num);
```

Comment:
this is a very interesting part... and your case should return undefined

the chrome result:
```
var a = new Int8Array(10);
>undefined
a[5]=50
>50
a[5.0]
>50
a["5"]
>50
a["5.0"]
>undefined
a["5e0"]
>undefined
```

Because the (spec 7.1.16)[http://www.ecma-international.org/ecma-262/6.0/#sec-canonicalnumericindexstring] use the string-number-string conversion to check if the prop name is CanonicalNumericIndexString. "5e1", "5.0" are not CanonicalNumericIndexString according to the criteria, and prop will be searched as normal way. e.g.

```
Int8Array.prototype["5e1"]="proto 5e1";
var a = new Int8Array(10);
a[5]=50
a["5e1"]
>"proto 5e1"
```